src/Evolution/DiscontinuousGalerkin/InitializeElement.hpp

```diff
@@ -0,0 +1,163 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/CreateInitialElement.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/Element.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/ElementIndex.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Time/Tags.hpp"
+#include "Time/Time.hpp"
+#include "Time/TimeId.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/Tuple.hpp"
+
+namespace dg {
+namespace Actions {
+/// \ingroup ActionsGroup
+/// \ingroup DiscontinuousGalerkinGroup
+/// \brief Initialize a dG element with analytic initial data
+///
+/// Uses:
+/// - ConstGlobalCache:
+///   * A tag deriving off of Cache::AnalyticSolutionBase
+///   * CacheTags::TimeStepper
+///
+/// DataBox changes:
+/// - Adds:
+///   * Tags::TimeId
+///   * Tags::Time
+///   * Tags::TimeStep
+///   * Tags::LogicalCoordinates<Dim>
+///   * Tags::Extents<Dim>
+///   * Tags::Element<Dim>
+///   * Tags::ElementMap<Dim>
+///   * System::variables_tag
+///   * Tags::HistoryEvolvedVariables<System::variables_tag,
+///                  db::add_tag_prefix<Tags::dt, System::variables_tag>>
+///   * Tags::Coordinates<Tags::ElementMap<Dim>,
+///                       Tags::LogicalCoordinates<Dim>>
+///   * Tags::InverseJacobian<Tags::ElementMap<Dim>,
+///                           Tags::LogicalCoordinates<Dim>>
+///   * Tags::deriv<System::variables_tag::tags_list,
+///                 System::gradients_tags,
+///                 Tags::InverseJacobian<Tags::ElementMap<Dim>,
+///                                       Tags::LogicalCoordinates<Dim>>>
+///   * db::add_tag_prefix<Tags::dt, System::variables_tag>
+///   * Tags::UnnormalizedFaceNormal<Dim>
+/// - Removes: nothing
+/// - Modifies: nothing
+template <size_t Dim>
+struct InitializeElement {
```

Comment:
Since this Action depends upon there being an analytic solution, do we want that reflected in the name of the Action?

Reply:
What my plans were was to make the Action detect if there's an analytic solution or not once we actually can read in other ID